### PR TITLE
Migrate aliased indices to data stream

### PR DESCRIPTION
**Pull request by danhermann**
Updates the `MetadataCreateDataStreamService` to permit existing indices as backing indices when creating a new data stream.

Relates to #61046

### CODE REVIEW

**Comment by danhermann** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by danhermann** [comment]
@martijnvg, I'm marking this as ready for review. I've moved the test code around to accommodate the new location of the validation logic. Let me know if I missed something you were expecting.

File: x-pack/plugin/data-streams/src/test/java/org/elasticsearch/xpack/datastreams/mapper/DataStreamTimestampFieldMapperTests.java [hunk 1]
```diff
@@ -277,17 +277,17 @@ public void testValidateNotDisallowedAttribute() throws IOException {
         assertThat(e.getMessage(), equalTo("data stream timestamp field [@timestamp] has disallowed attributes: [store]"));
     }
 
-    public void testCannotUpdateTimestampField() throws IOException {
+    public void testCanUpdateTimestampField() throws IOException {
```
Comment (martijnvg) [inline x-pack/plugin/data-streams/src/test/java/org/elasticsearch/xpack/datastreams/mapper/DataStreamTimestampFieldMapperTests.java:280]: I think we need to adjust test and the data stream timestamp field mapper.
Updating from `false` to `true` is allowed, but updating from `true` to `false` is disallowed.
`

Comment (martijnvg) [reply x-pack/plugin/data-streams/src/test/java/org/elasticsearch/xpack/datastreams/mapper/DataStreamTimestampFieldMapperTests.java:280]: This validation can be added by invoking `setMergeValidator(...)`method when building enabled parameter in `DataStreamTimestampFieldMapper` line 78.

**Review by martijnvg — COMMENTED** [review]
I left two comments, but other than that, this change is looking good!

**Comment by danhermann** [comment]
Thanks, @martijnvg. I've made those changes.

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by danhermann** [comment]
Thanks, @martijnvg!

**Pull request merged**